test/e2e/README.md

```diff
@@ -0,0 +1,363 @@
+# End-to-end test suite
+
+This is the entry point of the TriggerMesh E2E test suite.
+
+## Contents
+
+1. [Overview](#overview)
+1. [Running tests](#running-tests)
+   * [Execution](#execution)
+   * [Inputs](#inputs)
+1. [Package organization](#package-organization)
+1. [Writing tests](#writing-tests)
+   * [Structure](#structure)
+   * [Best practices](#best-practices)
+
+## Overview
+
+This package contains a single Go test called `TestE2e` which runs all the test specs contained in its sub-packages.
+
+Each test is written using the [Ginkgo][ginkgo-docs] testing framework using Behavior-Driven Development style ("BDD").
+
+The [`framework.Famework`](./framework/) abstraction, heavily inspired by [Kubernetes' E2E tests][k8s-e2e], offers a
+convenient way for each test spec to run against its own short-lived Kubernetes namespace, with common setup and cleanup
+tasks executed automatically before and after each of these specs.
+
+## Running tests
+
+### Execution
+
+| :warning: Do not ever interrupt running tests with `Ctrl-C`. When interrupted, Ginkgo only triggers the global `AfterSuite` logic and skips all the individual `AfterEach` blocks declared in running tests. This can leave dangling resources behind that are tedious to clean up manually and can become costly if not terminated quickly. Ref. [onsi/ginkgo#222][ginkgo-issue222].
+| :--- |
```

Comment:
We could add a troubleshooting section where we mention that all namespaces created for tests include the `e2e-framework` label (I think that's the case).

Reply:
Right, that would be a useful addition 👍 
`kubectl get ns -l e2e-framework`

Reply:
Done in [3ef0015a..55872614](https://github.com/triggermesh/triggermesh/compare/3ef0015a2ac139077c6f183985dfb8066b734bd2..558726140e0999d63cbbe2136fa31465a8e97a38)